indico_toolkit/staggered_loop/staggered_loop_recruiter.py

```diff
@@ -0,0 +1,174 @@
+        that particular field.
+
+        Args:
+            results (List[dict]): List of raw Indico result JSON
+            field (str): The particular label to query for
+        Returns:
+            pd.DataFrame: A Pandas DataFrame containing analysis of the results
+        """
+        df = pd.DataFrame()
+        for result in results:
+            workflow_result = WorkflowResult(result)
+            if workflow_result.post_reviews_predictions:
+                # Get Auto Review predictions
+                pre_review_preds = workflow_result.post_reviews_predictions[0]
+            else:
+                pre_review_preds = workflow_result.pre_review_predictions
+            final_preds = workflow_result.final_predictions
+
+            pre_review_pred_dict = pre_review_preds.to_dict_by_label
+            final_pred_dict = final_preds.to_dict_by_label
+
+            tp = [
+                pred
+                for pred in final_pred_dict[field]
+                if pred in pre_review_pred_dict[field]
+            ]
+            # Rejected predictions
+            fp = [
+                pred
+                for pred in pre_review_pred_dict[field]
+                if pred not in final_pred_dict[field]
+            ]
+            # Added predictions
+            fn = [
+                pred
+                for pred in final_pred_dict[field]
+                if pred not in pre_review_pred_dict[field]
+            ]
+
+            precision = len(tp) / (len(tp) + len(fp))
+            recall = len(tp) / (len(tp) + len(fn))
+
+            result_df = pd.DataFrame(
+                {
+                    "Precision": [precision],
+                    "Recall": [recall],
+                    "Rejected Predictions": [fp],
+                    "Added Predictions": [fn],
+                }
+            )
+            df = pd.concat([df, result_df], axis=0)
+
+        return df
+
+    def retrain(
+        self,
+        destination_client: IndicoClient,
+        dataset_id: int,
+        workflow_id: int,
+        teach_task_id: int,
+        files: List[str],
+        results: List[dict],
+        fields: List[str],
+    ):
+        """
+        For a given list of Indico results and desired fields, inject those results into prod model and retrain.
+
+        Args:
+            destination_client (IndicoClient): IndicoClient for the target instance to retrain on
+            dataset_id (int): The Id of the dataset to retrain on in the same instance as destination_client
+            workflow_id (int): The Id of the workflow to retrain on in the same instance as destination_client
+            teach_task_id (int): The Id of the teach task to retrain on in the same instance as destination_client
+            files (List[str]): List of file paths for corresponding documents you wish to inject labels into
+            results (List[dict]): List of results containing label information. files and results must map one-to-one with each other
```

Comment:
What is this? Is it the full result? Can this only be for 1 model extraction workflows?

Reply:
updated docstring to specify results as raw Indico model output